src/daemon/context.rs

```diff
@@ -38,13 +38,33 @@ pub static GLOBAL_IDENTITY: CtxField<IdentitySecret> = |ctx| {
             id.actualize()
                 .expect("failed to initialize global identity")
         })
-        .unwrap_or_else(IdentitySecret::generate)
+        .unwrap_or_else(|| {
+            let ctx = ctx.clone();
+            smol::future::block_on(async move {
+                match db_read(&ctx, "global_identity").await {
+                    Ok(Some(id)) => IdentitySecret::from_bytes(&id.try_into().unwrap()),
+                    _ => IdentitySecret::generate(),
+                }
+            })
+        })
 };
 
 pub static GLOBAL_ONION_SK: CtxField<OnionSecret> = |_| OnionSecret::generate();
-pub static RELAY_GRAPH: CtxField<RwLock<RelayGraph>> = |_| {
-    log::warn!("**** INIT RELAY GRAPH****");
-    RwLock::new(RelayGraph::new())
+pub static RELAY_GRAPH: CtxField<RwLock<RelayGraph>> = |ctx| {
+    let ctx = ctx.clone();
+    smol::future::block_on(async move {
+        match db_read(&ctx, "relay_graph").await {
+            Ok(Some(g)) => {
+                log::warn!("retrieving persisted relay graph");
+                let graph: RelayGraph = stdcode::deserialize(&g).unwrap();
```

Comment:
this should handle deserialization failures gracefully too

I think the easiest way is something like db_read(&ctx, "relay_graph").ok().flatten().and_then(|s| stdcode::deserialize(&s).ok()) which will give you one Option<RelayGraph> in one go